test/common/http/http2/codec_impl_test.cc

```diff
@@ -25,7 +26,25 @@ using testing::NiceMock;
 namespace Http {
 namespace Http2 {
 
-class Http2CodecImplTest : public testing::TestWithParam<uint64_t> {
+typedef ::testing::tuple<uint32_t, uint32_t, uint32_t, uint32_t> http2SettingsTuple;
+typedef ::testing::tuple<http2SettingsTuple, http2SettingsTuple> http2SettingsTupleTuple;
+
+static Http2Settings http2SettingsFromTuple(const http2SettingsTuple& tp) {
+  Http2Settings ret;
+  ret.hpack_table_size_ = ::testing::get<0>(tp);
+  ret.max_concurrent_streams_ = ::testing::get<1>(tp);
+  ret.initial_window_size_ = ::testing::get<2>(tp);
+  ret.initial_connection_window_size_ = ::testing::get<3>(tp);
+  return ret;
+}
+
+// skip test if server-side max concurrent streams is 0
+#define SKIP_TEST_IF_NOT_SUITABLE                                                                  \
```

Comment:
If you're going to make this zero (and I think it's fine to leave it as 1), then you probably do want to use gmock to verify that we get the right behavior here.